pagopa/io-app
# chore: [IOBP-977,IOBP-952] Add more payment explicit error and mixpanel events

## Short description
This PR handles the `PAYMENT_SLOWDOWN_ERROR` in a different way showing a different screen and emitting his custom mixpanel event and in case that there is any error not handled by default throws this generic error about the slowdown error.

## List of changes proposed in this pull request
- Added the `PaymentSlowdownErrorProblemJson` type with its enumerator `PAYMENT_SLOWDOWN_ERROR`;
- Mapped the backend GENERIC_ERROR and PAYMENT_VERIFY_GENERIC_ERROR with the mixpanel event `PAYMENT_502_ERROR`;
- Mapped the default code error when not found a mapped category with the `PAYMENT_NOT_MAPPED_CATEGORY_ERROR` code;
- Removed the condition inside the PickMethodScreen in order to show the correct error screen when happens any error to the requests of that screen instead navigating to the outcome screen;

## How to test
- Start a payment flow in local env;
- Simulate an error in the pick method screen simulating a 502 error or shutting down the dev-server and check that shows the correct error screen;
